server/src/main/java/org/opensearch/index/translog/transfer/TransferService.java

```diff
@@ -65,6 +66,7 @@ void uploadBlobs(
      * @throws IOException the exception while transferring the data
      */
     void uploadBlob(final TransferFileSnapshot fileSnapshot, Iterable<String> remotePath, WritePriority writePriority) throws IOException;
+    void uploadBlob(InputStream inputStream, Iterable<String> remotePath, String blobName, WritePriority writePriority, ActionListener<Void> listener) throws IOException;
```

Comment:
Shouldn't this be `uploadBlobAsync` ?

Reply:
Actually this method depends on the backing blob store to support async upload. Like S3 blob store supports async upload but FS blob store does not. So keeping the method name as uploadBlob only